core/src/alpenglow_consensus/certificate_pool.rs

```diff
@@ -595,10 +594,14 @@ pub(crate) fn load_from_blockstore(
             }));
 
         for (cert_id, cert) in certs {
-            let cert = cert.into_iter().map(Arc::from).collect();
             trace!("{my_pubkey}: loading certificate {cert_id:?} from blockstore into certificate pool");
-            let legacy_cert = LegacyVoteCertificate::new(cert_id, cert)
-                .expect("Certificate construction must not fail");
+            let mut legacy_cert = LegacyVoteCertificate::new(cert_id);
+            if let Err(e) = legacy_cert.aggregate(cert.iter()) {
+                error!(
+                    "{my_pubkey}: Error aggregating certificate {cert_id:?} from blockstore: {e:?}"
```

Comment:
nit: I would suggest panicking here as it means our blockstore is corrupted

Reply:
Done.